src/js/extension.js

```diff
@@ -0,0 +1,73 @@
+var Extension;
+(function(){
+
+    /* global Util */
+
+    Extension = function (options) {
+        Util.extend(this, options);
```

Comment:
fair enough, I guess. though -- the success of this shift (and all the regression work i've done) kind of relies on getting rid of the explicit "options" object for the individual things. the thinking there was: people are either passing new options or just wanting the defaults. by defining each extensions's "defaults" on the prototype you eliminate 1 level of nested objects, provide a mindlessly simple way to either pass options or extend the reference directly.  incidentially the root cause of me proposing any of this way the inconsistency in where options were read from and how they were passed, and this is kind of a pivotal moment/decision ... if the fundamental change of going from flat-options to nested-options also requires an explicit .options object to pass around nothing has been bettered and I should just stop this line of thinking all together (though once you see the end result it becomes way clearer)